### ISSUE TITLE
Add function to force a node offline.  

### ISSUE THREAD

**Issue by GarnetMonkey** [joel74/POSH-LTM-Rest#211]
Function ForceOffline-Node {
<#
.SYNOPSIS
    Force a node Offline to quickly disable all pool members associated with it
#>
    [cmdletBinding(DefaultParameterSetName='Address')]
    param (
        $F5Session=$Script:F5Session,

        [Alias('Node')]
        [Parameter(Mandatory,ParameterSetName='InputObject',ValueFromPipeline)]
        [PSObject[]]$InputObject,

        [Parameter(Mandatory,ParameterSetName='Address',ValueFromPipelineByPropertyName)]
        [Parameter(Mandatory,ParameterSetName='AddressAndName',ValueFromPipelineByPropertyName)]
        [PoshLTM.F5Address[]]$Address=[PoshLTM.F5Address]::Any,

        [Alias('ComputerName')]
        [Alias('NodeName')]
        [Parameter(Mandatory,ParameterSetName='Name',ValueFromPipeline,ValueFromPipelineByPropertyName)]
        [Parameter(Mandatory,ParameterSetName='AddressAndName',ValueFromPipeline,ValueFromPipelineByPropertyName)]
        [string[]]$Name='',

        [Parameter(ValueFromPipelineByPropertyName)]
        [string]$Partition,

        [switch]$Force
    )
    begin {
        #If the -Force param is specified pool members do not accept any new connections, even if they match an existing persistence session.
        #Otherwise, members will only accept new connections that match an existing persistence session.
    }
    process {
        switch($PSCmdLet.ParameterSetName) {
            InputObject {
                If ($Force){
                    $AcceptNewConnections = "user-down"
                }
                Else {
                    $AcceptNewConnections = "user-up"
                }
                $JSONBody = @{state=$AcceptNewConnections;session='forced_offline'} | ConvertTo-Json
                foreach($member in $InputObject) {
                    $URI = $F5Session.GetLink($member.selfLink)
                    Invoke-F5RestMethod -Method PATCH -Uri "$URI" -F5Session $F5Session -Body $JSONBody -ErrorMessage "Failed to force node offline $($member.Name)." -AsBoolean
                }
            }
            default {
                Get-Node -F5session $F5Session -Address $Address -Name $Name -Partition $Partition | Disable-Node -F5session $F5Session -Force:$Force
            }
        }
    }
}

**Comment by joel74** [comment]
Thanks, @GarnetMonkey . As I look at this, I think it makes the most sense to combine this functionality with the existing Disable-Node function, and add a switch for the option to set the session to either 'forced_offline' or 'user-disabled.'
I'll work on that.

**Comment by jpanda11** [comment]
Hello Dear Joel, 
was just wondering if it is possible to force offline node yet? 
Thank you very much :)

**Comment by joel74** [comment]
Hi, @jpanda11 , I'm not completely clear on what the difference is between:
state='user-down'
and
session='forced_offline'

In the current version of Disable-Node, when I pass the -Force switch, the call sets "state='user-down'"
When I look at the node in the UI, it appears as Offline (Disabled) - Forced Down.

When I changed the value of session in the call to 'forced_offline,' my call to disable the node with the -Force switch returned false.

**Comment by joel74** [comment]
@jpanda11 , can you please let me know if this resolved your issue? @GarnetMonkey , can you let me know if using the -force switch when calling Disable-Node achieves the desired result? Thanks.

**Comment by jpanda11** [comment]
@joel74 Hi,
I was able to take the node in "Forced Offline" mode using the -Force switch.

The difference is, 
Force-Offline - Closes every connection and no new connection to the server is allowed, whereas,
Disabled - No new connections are allowed but keep them connected session alive unless disconnected by itself.

Our requirement for some webservers is Forced-Offline and for others Disabled. I have been able to achieve both with the Disable-Node command :)
Thank you very much :)
